pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -445,8 +455,14 @@ func (p *provider) Validate(ctx context.Context, spec clusterv1alpha1.MachineSpe
 	if err != nil {
 		return fmt.Errorf("failed to parse config: %w", err)
 	}
-	// If VMIPreset is specified, skip CPU and Memory validation.
-	if c.FlavorName == "" {
+	// If instancetype is specified, skip CPU and Memory validation.
+	// Values will come from instancetype.
+	if c.Instancetype == nil {
```

Comment:
What if we combine:
```
if c.Instancetype == nil && c.FlavorName == "" {
...
}
```
As inside both conditions, it is doing the same `pasreResource`?